test/rules/assertions/validTypes.js

```diff
@@ -161,14 +161,82 @@ export default {
           }
       `,
       errors: [{
-        line: 3,
-        message: 'Syntax error in type: '
+        line: 2,
+        message: 'Tag @callback must have a namepath'
       }],
       settings: {
         jsdoc: {
           allowEmptyNamepaths: false
         }
       }
+    },
+    {
+      code: `
+          /**
+           * @constant {str%ng}
+           */
+           const FOO = 'foo';
+      `,
+      errors: [
+        {
+          line: 3,
+          message: 'Syntax error in type: str%ng'
+        }
+      ]
+    },
+    {
```

Comment:
Since `constant` allows an optional name per https://jsdoc.app/tags-constant.html, maybe you could also add a test for `@constant {type} name`?

Reply:
Added. 

But I'm not entirely sure how that form is supposed to be used. What if I have different names in tag and variable declaration?

Reply:
I think, like `check-param-names`, name-checking could be useful within a new rule like `check-const-names` (or something more generic to work with `@member`/`@memberof!`, `@var`, and possibly `@namespace` as well as `@constant`/`@const`). However, for `valid-types`, I think just ensuring the rule doesn't break when a name is present would be fine.

Reply:
So, I'll mark this as resolved, as it seems you've already added such an example.